Documentation/components/filesystem/littlefs.rst

```diff
@@ -4,3 +4,26 @@ LITTLEFS
 
 A little fail-safe filesystem designed for microcontrollers from
 https://github.com/littlefs-project/littlefs.
+
+In NuttX, littlefs can be interacts with through the virtual file system. This
+means that it can be used normally with ``read()``, ``write()``, etc. calls, as
+well as file operations (``fopen()``, ``fclose()``, etc).
+
+.. note::
+
+   Since littlefs is power-fail safe and must commit writes before they are
+   permanently stored (corruption prevention mechanism), it is required to
+   periodically call ``fsync`` on the file to commit your writes. The exact
+   semantics of when littlefs files are committed are discussed in `this issue
+   <https://github.com/apache/nuttx/issues/15840>`_.
+
+
+..note:: 
+
+  If your littlefs setup is experiencing crashes when you boot, try
+  troubleshooting by tweaking the ``BLOCK_SIZE_FACTOR`` options in Kconfig.
```

Comment:
what is this note about?

Reply:
On SD cards, a block size of 4 prevents a kernel panic. This is not the default block size, which I assume is better suited for flash devices.